plugins/database/src/db_plugin.cpp

```diff
@@ -14529,13 +14529,283 @@ irods::error db_get_hierarchy_for_resc_op(
+                                                   const KeyValPair* _cond_input)
+    {
+        //
+        // Get the ticket's id.
+        //
+
+        rodsLong_t ticket_id = 0;
+        std::vector<std::string> bindVars{_ticket_string};
+
+        auto ec = cmlGetIntegerValueFromSql("select ticket_id from R_TICKET_MAIN where ticket_string = ?",
+                                            &ticket_id, bindVars, &icss);
+
+        if (0 != ec) {
+            ec = cmlGetIntegerValueFromSql("select ticket_id from R_TICKET_MAIN where ticket_id = ?",
+                                           &ticket_id, bindVars, &icss);
+
+            if (0 != ec) {
+                return ERROR(CAT_TICKET_INVALID, _ticket_string);
+            }
+        }
+
+        const auto ticket_id_string = std::to_string(ticket_id);
+
+        //
+        // Handle the operation.
+        //
+
+        // Delete ticket operation.
+        if (std::strcmp(_op_name, "delete") == 0) {
+            cllBindVars[0] = ticket_id_string.c_str();
+            cllBindVarCount = 1;
+
+            //
+            // Delete ticket from primary ticket table.
+            //
+            
+            ec = cmlExecuteNoAnswerSql("delete from R_TICKET_MAIN where ticket_id = ?", &icss);
+            if (CAT_SUCCESS_BUT_WITH_NO_INFO == ec) {
+                return CODE(ec);
+            }
+
+            if (0 != ec) {
+                return ERROR(ec, fmt::format("Failed to delete ticket with id [{}] as user [{}].",
+                                             ticket_id_string, _ctx.comm()->clientUser.userName));
+            }
+
+            //
+            // Delete all relationships stored in the secondary ticket tables.
+            //
+
+            cllBindVars[0] = ticket_id_string.c_str();
+            cllBindVarCount = 1;
+            ec = cmlExecuteNoAnswerSql("delete from R_TICKET_ALLOWED_HOSTS where ticket_id = ?", &icss);
+            if (0 != ec && CAT_SUCCESS_BUT_WITH_NO_INFO != ec) {
+                logger::database::warn("Failed to delete ticket information [error_code={}, table=R_TICKET_ALLOWED_HOSTS]", ec);
+            }
+
+            cllBindVars[0] = ticket_id_string.c_str();
+            cllBindVarCount = 1;
+            ec = cmlExecuteNoAnswerSql("delete from R_TICKET_ALLOWED_USERS where ticket_id = ?", &icss);
+            if (0 != ec && CAT_SUCCESS_BUT_WITH_NO_INFO != ec) {
+                logger::database::warn("Failed to delete ticket information [error_code={}, table=R_TICKET_ALLOWED_USERS]", ec);
+            }
+
+            cllBindVars[0] = ticket_id_string.c_str();
+            cllBindVarCount = 1;
+            ec = cmlExecuteNoAnswerSql("delete from R_TICKET_ALLOWED_GROUPS where ticket_id = ?", &icss);
+            if (0 != ec && CAT_SUCCESS_BUT_WITH_NO_INFO != ec) {
+                logger::database::warn("Failed to delete ticket information [error_code={}, table=R_TICKET_ALLOWED_GROUPS]", ec);
```

Comment:
Should these include the ticket id, or does that not matter at this point?

Reply:
Not a bad idea. Will have to change in the main branch too.

Reply:
done. will port to the main branch too.

Reply:
done.